src/format.ts

```diff
@@ -75,217 +42,57 @@ export async function format(
  */
-function fixFormat(srcFiles: string[], baseArgs: string[]): Promise<boolean> {
-  return new Promise<boolean>((resolve, reject) => {
-    const args = baseArgs.concat(['-i'], srcFiles);
-    clangFormat.spawnClangFormat(
-      args,
-      (err?: Error) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(true);
-        }
-      },
-      'inherit'
-    );
-  });
-}
-
-/**
- * Runs clang-format on the list of files and checks whether they are formatted
- * correctly. Returns true if all files are formatted correctly.
- *
- * @param srcFiles list of source files
- */
-function checkFormat(
-  options: Options,
-  srcFiles: string[],
-  baseArgs: string[]
-): Promise<boolean> {
-  return new Promise<boolean>((resolve, reject) => {
-    let output = '';
-    const args = baseArgs.concat(['-output-replacements-xml'], srcFiles);
-    const out = clangFormat.spawnClangFormat(
-      args,
-      (err?: Error) => {
-        if (err) {
-          reject(err);
-        }
-      },
-      ['ignore', 'pipe', process.stderr]
-    ).stdout;
-    out.setEncoding('utf8');
-    out.on('data', (data: Buffer) => {
-      output += data;
-    });
-
-    out.on('end', async () => {
-      const files: string[] = output.split("<?xml version='1.0'?>\n").slice(1);
-      for (let i = 0; i < files.length; i++) {
-        if (files[i].indexOf('<replacement ') === -1) {
-          continue;
-        }
-        const replacements = getReplacements(files[i]);
-        if (replacements.length > 0) {
-          const diff = await getDiffObj(srcFiles[i], replacements);
-          printDiffs(diff, options);
-        }
-      }
-      resolve(output.indexOf('<replacement ') === -1 ? true : false);
-    });
-  });
-}
-
-/**
- * Parses through xml string for the replacement string, offset, length and
- * returns all of the necessary replacements for the file
- *
- * @param output xml string from clangFormat
- */
-export function getReplacements(fileXML: string): Replacement[] {
-  const replacements: Replacement[] = [];
-
-  let xmlLines = fileXML.trim().split('\n');
-  // first and last elements are outer 'replacements' tags
-  xmlLines = xmlLines.slice(1, xmlLines.length - 1);
-
-  for (let i = 0; i < xmlLines.length; i++) {
-    // Uses regex to capture the xml attributes and element
-    // XML format:
-    // <replacement offset='OFFSET' length='LENGTH'>FIX</replacement>
-    const offset: string[] | null = /offset='(\d+)'/g.exec(xmlLines[i]);
-    const length: string[] | null = /length='(\d+)'/g.exec(xmlLines[i]);
-    const fix: string[] | null = /length='\d+'>(.*)<\/replacement>/g.exec(
-      xmlLines[i]
-    );
-
-    if (length === null || offset === null || fix === null) {
-      throw new Error('Unable to get replacement');
+async function checkFormat(srcFiles: string[], options: Options) {
+  const configs = await Promise.all(srcFiles.map(file =>
+      prettier.resolveConfig(file)));
```

Comment:
One concern I have here is the # of different formats prettier supports for config files:
https://prettier.io/docs/en/configuration.html

Does using this method mean that we'll only drop a `prettier.config.js` if none of the 4 options are present?  Or is there potential for us to clobber on init?

Reply:
This should look for any of the variants of the config files that `prettier` supports. If none is found, it will use the default from `gts`.

Reply:
Added a test for this so that you don't have to take my word for it.